addons/website/models/website_page.py

```diff
@@ -61,6 +62,27 @@ def _compute_visible(self):
                 not page.date_publish or page.date_publish < fields.Datetime.now()
             )
 
+    @api.depends('menu_ids')
+    def _compute_website_menu(self):
+        for page in self:
+            page.is_in_menu = bool(page.menu_ids)
+
+    def _set_website_menu(self):
+        for page in self:
+            website = self.env['website'].get_current_website()
```

Comment:
You're right, my bad! I ignored the fact that we don't edit a page in the context of its related website.
e.g. if we set `is_in_menu=True` on page form view (page website_id = W1) while current website is W2 => this will create the menu on W2 > inconsistent). **[*]**
Thanks for spotting this :+1: